src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private double length;
+    private double width;
+
+    public Rectangle(String color, double length, double width) {
+        super(color);
+        this.length = length;
+        this.width = width;
+    }
+
+    public double getLength() {
+        return length;
+    }
+
+    public double getWidth() {
+        return width;
+    }
+
+    @Override
+    public double getArea() {
+        return length * width;
+    }
+
+    @Override
+    public String draw() {
+        return "Figure: rectangle, area: "
+                + String.format("%.2f",getArea()) + " sq. units, length: "
+                + String.format("%.2f",getLength()) + " units, width: "
+                + String.format("%.2f",getWidth()) + " units, color: " + getColor();
```

Comment:
The `draw()` method should not be included in the same interface or abstract class as `getArea()` since they are conceptually different. Consider separating these responsibilities into different interfaces or abstract classes.